### PR TITLE
pep and ruff cleanup in semirings and tropicals

### PR DESCRIPTION
**Pull request by fchapoton**
some code cleanup (pep8 and ruff suggestions) in rings/semirings/

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: src/sage/rings/semirings/tropical_polynomial.py [hunk 1]
```diff
@@ -383,13 +383,13 @@ def piecewise_function(self):
         if len(data) == 1:
             gradient = list(data)[0]
             intercept = data[gradient].lift()
-            f = intercept + gradient*x
+            f = intercept + gradient * x
```
Comment (gmou3) [inline src/sage/rings/semirings/tropical_polynomial.py]: I think this change goes contrary to PEP8:
https://peps.python.org/pep-0008/#whitespace-in-expressions-and-statements.

Comment (fchapoton) [reply src/sage/rings/semirings/tropical_polynomial.py]: Maybe, indeed. Do you want me to undo this ?

Comment (gmou3) [reply src/sage/rings/semirings/tropical_polynomial.py]: Yes, please. Otherwise looks good.

Comment (fchapoton) [reply src/sage/rings/semirings/tropical_polynomial.py]: done

**Review by gmou3 — APPROVED** [review]

**Pull request merged**